### PR TITLE
Add Forms

### PR DESCRIPTION
**Pull request by PraisesPJMT**
# Inkling Blog App | Add Forms #7
## Project Description
Use forms to add new `Post`, `Comment`, and `Like` for the `Inkling Blog` app. 
## Feature Description
-  Created a method called `#current_user` in the ApplicationController to make current user data available to all controllers.
    -  `#current_user` returns the first user from the database.
- Created forms to perform the following functions:
    - Creates a new `Post` on behalf of the `current_user`.
    - Creates a `Comment` on behalf of the `current_user` .
- Allow User (`current_user`) to `Like` a `Post`.
## Recommendations
An optional boilerplate of data entry for testing the app is contained in the `db/seed.rb`. Run the following commands in sequence to be able to use the boilerplate.
```sh
# Create Database
rails db:create
```

```sh
# Generate Database Tables
rails db:migrate
```

```sh
# Use Boilerplates to Populate Tables
rails db:seeding
```

```sh
# Use Boilerplates to Populate Tables
rails server
```

## Preview
**New Posts**
Web capture_9-12-2022_175811_127 0 0 1

**Comments and Likes**
Web capture_9-12-2022_174243_127 0 0 1

Happy Reviewing!!!!!

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+    @post = Post.find(params[:post_id])
```
Comment (Deepakdanger) [inline app/controllers/comments_controller.rb]: - [x] Why returning @post in comments new. Kindly remove it

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      flash[:error] = 'There was a problem creating the comment'
+    end
+    redirect_to "/users/#{author.id}/posts/#{post.id}"
+  end
+
+  # private
+  #
+  # def comment_params
+  #   params.require(:comment).permit(:text)
+  # end
```
Comment (Deepakdanger) [inline app/controllers/comments_controller.rb]: - [x] Kindly use this strong parameter at the time of comment creation in line 10. as `comment_params`

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -7,4 +7,26 @@ def show
...
   end
+
+  def new
+    @post = Post.new
+
+    render :new, locals: { post: @post }
+  end
+
+  def create
+    post = current_user.posts.new(params.require(:post).permit(:title, :text))
```
Comment (Deepakdanger) [inline app/controllers/posts_controller.rb]: -[ ] Kindly use strong parameters here while creating posts for more security. 
```
 private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```
& use `post_params` in line 18.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @PraisesPJMT , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (PraisesPJMT) [reply app/controllers/comments_controller.rb]: I wanted using the `@post.id` for routing. I later didn't us it so it is actually redundant as you have observed. Thanks @deepakdanger for the recommendation!

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (PraisesPJMT) [reply app/controllers/comments_controller.rb]: Strong parameter is now added to comment creation! Thanks @Deepakdanger for the recommendation!

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (PraisesPJMT) [reply app/controllers/posts_controller.rb]: Strong parameter is now added to post creation! Thanks @Deepakdanger for the recommendation!

**Comment by PraisesPJMT** [comment]
## Review Feedback
Thanks @Deepakdanger for the recommendations. The following changes were made based on the review:

### Review Changes
- Strong parameter was added and used for the creation of `Post`.
- Strong parameter was added and used for the creation of `Comment`.
- Redundant `@post` instance variable was remove from `#new` method in `Comment`  `ActionController`.


Happy Reviewing!!!!!

**Comment by arslanbisharat** [comment]
Hi @PraisesPJMT,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by PraisesPJMT** [comment]
Thanks so much @arslanbisharat  for the feedback! I appreciate your amazing review!

**Pull request merged**